xyzspaces/spaces.py

```diff
@@ -1326,3 +1326,39 @@ def add_features_geopandas(
                 features_size=features_size,
                 chunk_size=chunk_size,
             )
+
+    def clone(self, space_id: str = None, limit: int = 1000):
+        """
+        Copy current space data into a new space.
+
+        :param space_id: space id into which to copy data,
+            if not provided will create a new space and copy the data.
+        :param limit: A max. number of features to read in a single iteration
+            while iterating over the source space.
+        :return: The cloned Space Object
+        """
+        if space_id:
+            clone_space = Space.from_id(space_id=space_id)
+        else:
+            title = self.info["title"]
+            desc = self.info["description"]
+            clone_space = Space.new(title=title, description=desc)
+        features = []
+        feature_collection: Dict[Any, Any] = {}
+        for f in self.iter_feature(limit=limit):
+            features.append(f)
+            if len(features) == limit:
+                feature_collection = dict(
+                    type="FeatureCollection", features=features
+                )
+                clone_space.add_features(features=feature_collection)
+                features = []
+                feature_collection = {}
+
+        if len(features) != 0:
```

Comment:
Really `!= 0`, can it be negative? Better: `> 0`.

Reply:
Iterate features can read >=0 features only

Reply:
That's why.